### PR TITLE
Chore/dw 511

### PR DESCRIPTION
**Pull request by martincousido**

### CODE REVIEW

File: data-weave-lang-server/src/main/scala/org/mule/weave/lsp/WeaveLanguageServer.scala [hunk 1]
```diff
@@ -235,7 +235,10 @@ class WeaveLanguageServer extends LanguageServer {
 
   @JsonNotification("weave/project/create")
   def createProject(): Unit = {
-    val projectProvider = new ProjectProvider(client, workspaceUri)
-    projectProvider.newProject()
+    CompletableFuture.supplyAsync(() => {
```
Comment (machaval) [inline data-weave-lang-server/src/main/scala/org/mule/weave/lsp/WeaveLanguageServer.scala:238]: Aren't we going to change this for a command?

File: data-weave-lang-server/src/main/scala/org/mule/weave/lsp/project/commands/ProjectProvider.scala [hunk 2]
```diff
@@ -23,36 +25,71 @@ class ProjectProvider(client: WeaveLanguageClient, workspaceLocation: URI) {
...
-      askForName("example-project", "Project name").mapOptionInside { projectName => (organizationId, artifactId, version, projectName) }
-    }.flatMapOption { case (organizationId, artifactId, version, projectName) =>
-      askForPath(Some(new File(workspaceLocation).toPath)).mapOptionInside { projectPath =>
-        val projectUri = projectPath.toUri
-        new ProjectCreator(ProjectCreationInfo(organizationId, artifactId, version, projectName, projectUri)).create()
-        askForWindow(projectUri.resolve(projectName))
-      }
-    }
+
+    val createProjectTitle = "Create Project"
```
Comment (machaval) [inline data-weave-lang-server/src/main/scala/org/mule/weave/lsp/project/commands/ProjectProvider.scala:29]: Very nice DSL

**Review by machaval — APPROVED** [review]

File: data-weave-lang-server/src/main/scala/org/mule/weave/lsp/ui/wizard/QuickPickWidgetBuilder.scala [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    this
+  }
+}
+
+class QuickPickWidget[T](languageClient: WeaveLanguageClient, quickInputWidgetBuilder: QuickPickWidgetBuilder[T]) extends Widget[T] {
+
+  var buttons: Map[String, (T => WidgetResult[T])] = Map()
+  var items: Map[String, (T => WidgetResult[T])] = Map()
+
+  override def show(resultToFill: T): WidgetResult[T] = {
```
Comment (martincousido) [inline data-weave-lang-server/src/main/scala/org/mule/weave/lsp/ui/wizard/QuickPickWidgetBuilder.scala:67]: Need to refactor these, a lot of common stuff between this one and the InputWidget

File: data-weave-lang-server/src/main/scala/org/mule/weave/lsp/WeaveLanguageServer.scala (hunk 1 above)
Comment (martincousido) [reply data-weave-lang-server/src/main/scala/org/mule/weave/lsp/WeaveLanguageServer.scala:238]: Yes, I'll change this next

File: data-weave-lang-server/src/main/scala/org/mule/weave/lsp/project/commands/DefaultProjectCreator.scala [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import java.io.PrintWriter
+import java.nio.file.Files
+import java.nio.file.Path
+import scala.io.Source._
+
+
+class DefaultProjectCreator() extends ProjectCreator {
+
+  def create(projectInfo: ProjectCreationInfo): Unit = {
+    val source = fromResource("pom-default.xml")
```
Comment (martincousido) [inline data-weave-lang-server/src/main/scala/org/mule/weave/lsp/project/commands/DefaultProjectCreator.scala:13]: Need to close this resource :'(

**Pull request merged**